WireUI/Sources/WireMainNavigationUI/Coordinator/MainCoordinator.swift

```diff
@@ -197,24 +193,26 @@ public final class MainCoordinator<Dependencies>: NSObject, MainCoordinatorProto
     public func showConversation(
         conversation: ConversationModel,
         message: ConversationMessageModel?
-    ) async {
-        if mainSplitViewState == .expanded, splitViewController.splitBehavior == .overlay {
-            splitViewController.hideSidebar()
-        }
-
-        await dismissPresentedViewController()
+    ) {
+        Task { @MainActor in
+            if mainSplitViewState == .expanded, splitViewController.splitBehavior == .overlay {
+                splitViewController.hideSidebar()
+            }
 
-        let conversationUI = conversationUIBuilder.build(
-            conversation: conversation,
-            message: nil,
-            mainCoordinator: self
-        )
-        if mainSplitViewState == .collapsed {
-            tabBarController.selectedContent = .conversations
-            await Task.yield() // without this line subsequent navigation controller push animations don't work
-            tabBarController.setConversationUI(conversationUI, animated: true)
-        } else {
-            splitViewController.conversationUI = conversationUI
+            await dismissPresentedViewController()
+
+            let conversationUI = conversationUIBuilder.build(
+                conversation: conversation,
+                message: nil,
+                mainCoordinator: self
+            )
+            if mainSplitViewState == .collapsed {
+                tabBarController.selectedContent = .conversations
+                await Task.yield() // without this line subsequent navigation controller push animations don't work
```

Comment:
I think because we reset the stack of the navigation controller the view's need to be laid out again.
Maybe a `setNeedsLayout()` + `layoutIfNeeded` would work as well.